Various small tweaks to mypy daemon documentation (#8041)
python/mypy: docs/source/mypy_daemon.rst

   Use ``FILE`` as the status file for storing daemon runtime state. This is
   normally a JSON file that contains information about daemon process and
   connection. The default path is ``.dmypy.json`` in the current working
   directory.

.. option:: --log-file FILE

   Direct daemon stdout/stderr to ``FILE``. This is useful for debugging daemon
   crashes, since the server traceback is not always printed by the client.
   This is available for the ``start``, ``restart``, and ``run`` commands.

.. option:: --timeout TIMEOUT

   Automatically shut down server after ``TIMEOUT`` seconds of inactivity.
   This is available for the ``start``, ``restart``, and ``run`` commands.

.. option:: --update FILE

   Re-check ``FILE``, or add it to the set of files being
   checked (and check it). This option may be repeated, and it's only available for
   the ``recheck`` command.  By default, mypy finds and checks all files changed

>    since the previous run.  However, if you use this option

Reviewer: Maybe add something like this:
```suggestion
   since the previous run and files that depend on them.  However, if you use this option
```

   (and/or :option:`--remove`), mypy assumes that only the explicitly

>    specified files need to be reprocessed. This is only useful to

Reviewer: ```suggestion
   specified files have changed. This is only useful to
```

   speed up mypy if you type check a very large number of files, and use an
   external, fast file system watcher, such as `watchman`_ or

>    `watchdog`_, to determine which files need to be type checked.

Reviewer: Also here:
```suggestion
   `watchdog`_, to determine which files got edited or deleted.
```
